### PR TITLE
Pull Request for Creating Association and Queries

### PR DESCRIPTION
**Pull request by chuaindia**
In this PR, the following features are incorporated : 

- Some additional methods are added.
- For "Users", a method is created that returns the 3 most recent posts for a given user.
- For "Posts", a method is created that updates the posts counter for a user.
- For "Posts", a method is created which returns the 5 most recent comments for a given post.
- For "Comments", a method is created that updates the comments counter for a post.
- For "Likes", a method is created that updates the likes counter for a post.
- This model is used to insert data in the console.

### CODE REVIEW

File: app/models/post.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  def updates_the_posts_counter
+    user.increment!(:PostCounter)
+  end
+
+  private :updates_the_posts_counter
+
+  validates :Title, presence: true, length: { maximum: 250 }
+  validates :CommentCounter, numericality: { greater_than_or_equal_to: 0 }
+  validates :LikeCounter, numericality: { greater_than_or_equal_to: 0 }
```
Comment (thecodechaser) [inline app/models/post.rb]: - [x] Please remove the validations, You will implement validations in the upcoming project.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  has_many :posts, foreign_key: 'AuthorId'
+  has_many :comments, foreign_key: 'AuthorId'
+  has_many :likes, foreign_key: 'AuthorId'
+
+  def most_recent_posts
+    Post.where(AuthorId: id).last(3)
+  end
+
+  validates :Name, presence: true
+  validates :PostCounter, numericality: { greater_than_or_equal_to: 0 }
```
Comment (thecodechaser) [inline app/models/user.rb]: - [x] Please remove the validations, You will implement validations in the upcoming project.

**Review by thecodechaser — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @chuaindia👋, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight 🎉
- No linter errors. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️



## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/post.rb (hunk 1 above)
Comment (chuaindia) [reply app/models/post.rb]: Ok , I am removing that.

File: app/models/user.rb (hunk 2 above)
Comment (chuaindia) [reply app/models/user.rb]: Ok, I am removing that.

File: app/models/post.rb (hunk 1 above)
Comment (chuaindia) [reply app/models/post.rb]: Done.

File: app/models/user.rb (hunk 2 above)
Comment (chuaindia) [reply app/models/user.rb]: Done.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi @chuaindia, 

### Good job !!!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  Good commit messages
:heavy_check_mark:  Clean and organized code
:heavy_check_mark: Descriptive Readme file

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear or you can get to me on slack  using this handle @mihndim.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**